src/dataset/dataset_coco.py

```diff
@@ -312,3 +316,41 @@ def extract_patches(self, output_dir: str, patch_size: int, stride: int, min_are
                 pass
 
         patch_annotations.save(output_path=os.path.join(output_dir, "annotations", "annotations.json"))
+
+
+def extended_dimensions(patch_size: int, stride: int, image_shape: Tuple[int, ...]) -> Tuple[int, ...]:
+    # TODO test cases with strides greater than patch_size
+    return [
+        image_shape[i]
+        if (image_shape[i] - patch_size) % stride == 0
+        else (((image_shape[i] // stride) * stride) + patch_size)
+        for i in range(len(image_shape))
+    ]
+
+    # L = 9; P = 3; S = 4
+    # o o o o X X X X X  X X X o o o o X X  X X X X X X o o o o
+    # o o o o X X X X X  X X X o o o o X X  X X X X X X o o o o
+    # o o o o X X X X X  X X X o o o o X X  X X X X X X o o o o
+    # o o o o X X X X X  X X X o o o o X X  X X X X X X o o o o
+    # X X X X X X X X X  X X X X X X X X X  X X X X X X X X X
+    # Need to add 1 column.
+
+    # L = 11; P = 4; S = 3
+    # o o o o X X X X X X X  X X X o o o o X X X X  X X X X X X o o o o X  X X X X X X X X X o o o o
+    # o o o o X X X X X X X  X X X o o o o X X X X  X X X X X X o o o o X  X X X X X X X X X o o o o
+    # o o o o X X X X X X X  X X X o o o o X X X X  X X X X X X o o o o X  X X X X X X X X X o o o o
+    # o o o o X X X X X X X  X X X o o o o X X X X  X X X X X X o o o o X  X X X X X X X X X o o o o
+    # X X X X X X X X X X X  X X X X X X X X X X X  X X X X X X X X X X X  X X X X X X X X X X X
+    # Need to add 2 columns.
+
+    # L = 10; P = 4; S = 2
+    # o o o o X X X X X X  X X o o o o X X X X  X X X X o o o o X X  X X X X X X o o o o
+    # o o o o X X X X X X  X X o o o o X X X X  X X X X o o o o X X  X X X X X X o o o o
+    # o o o o X X X X X X  X X o o o o X X X X  X X X X o o o o X X  X X X X X X o o o o
+    # o o o o X X X X X X  X X o o o o X X X X  X X X X o o o o X X  X X X X X X o o o o
+    # X X X X X X X X X X  X X X X X X X X X X  X X X X X X X X X X  X X X X X X X X X X
+    # No need to add columns
+
+    # L = 9; P = 3; S = 2
+    # o o o X X X X X X  X X o o o X X X X  X X X X o o o X X  X X X X o o o X X  X X X X X X o o o
+    # No need to add columns
```

Comment:
Is this still necessary?

Reply:
Nope, was there so we could discuss/reason.